frontends/web/src/routes/new-settings/utils.ts

```diff
@@ -0,0 +1,34 @@
+import i18n from '../../i18n/i18n';
+
+export type TTab = {
+    url: string;
+    name: string;
+}
+
+const settingsTabsDetail: TTab[] = [
+  { url: '/new-settings/appearance', name: i18n.t('settings.appearance') },
+  { url: '/new-settings/manage-accounts', name: 'Manage accounts' },
+  { url: '/new-settings/advanced-settings', name: 'Advanced settings' },
+  { url: '/new-settings/about', name: 'About' },
+];
+
+export const settingsTabsBuilder = (deviceIDs: string[]) => {
+  if (deviceIDs.length >= 1) {
+    const deviceSettingsTabsDetail = deviceIDs.map(id => ({ url: `/device/${id}`, name: 'Device settings' }));
+
+    const settingsTabsWithDeviceSettings = [
+      // keep the first 2 indices
+      ...settingsTabsDetail.slice(0, 2),
+      // insert the device settings tab(s)
+      // between the first 2 indices and the rest of the tabs
+      ...deviceSettingsTabsDetail,
+      // the rest of the tabs
+      // after all of the device setting tab(s)
+      ...settingsTabsDetail.slice(2)
+    ];
+
+    return settingsTabsWithDeviceSettings;
+  }
+
+  return settingsTabsDetail;
+};
```

Comment:
ignoring how the component is built right now, I'd slightly prefer something like this:

```jsx
export const settingsTabsBuilder = (deviceIDs: string[]) => {
  return (
    <>
      <tab url="/new-settings/appearance">{t('settings.appearance')}</tab>
      <tab url="/new-settings/manage-accounts">Manage accounts</tab>
      {deviceIDs.map(id => (
        <tab url={`/device/${id}`}>Device settings</tab>
      )) }
      <tab url="/new-settings/advanced-settings">Advanced settings</tab>
      <tab url="/new-settings/about">About</tab>
    </>
  );
};
```

Reply:
I was in a hurry a bit when reviewing before. there is also condition to render "Manage accounts" only when there is an unlocked BB02. But not a blocker can be done later.